File: codes/javascript/encapsulation/person.js
module.exports = Person;

function Person(name) {
    this.luckyNumber = 0;
    this._name = name;
    var num = Person.nbr;
    Person.nbr++;

    this.info1 = function(){
        console.log("My number is: ", num);
    }
}

Person.nbr = 0;

Person.prototype.info = function(){
    console.log("My name: ", this._name);
    this.info1();
    console.log("My lucky number is: ", luckyNumber);
    console.log("--------------------------");
}

File: codes/python/encapsulation/person.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Person(object):

	nbr = 0

	def __init__(self, name):
		self.__name = name
		self.__num = Person.nbr
		self._t = "person"
		self.luckyNumber = 0
		Person.nbr += 1

	def info(self):
		print("My name: " + self.__name)
		print("My number: " + str(self.__num))
		print("I am a: " + self._t)
		print("My lucky number is: " + str(self.luckyNumber))
		print("--------------------------")
